gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -926,11 +941,11 @@ public void run() {
           for (Map<Object, Object> regionEntryMap : oldEntryMap.values()) {
             for (Entry<Object, Object> entry : regionEntryMap.entrySet()) {
               if (entry.getValue() instanceof BlockingQueue) {
-                if (((BlockingQueue) entry.getValue()).isEmpty()) {
-                  regionEntryMap.remove(entry.getKey());
+                if (((BlockingQueue)entry.getValue()).isEmpty()) {
```

Comment:
This isEmpty check is not required since it is handled by the queueRemover already. Also can have some rare race condition where queue becomes empty just after the check so better to do in the segment lock as taken by remove call below.

Reply:
My thinking was to reduce contention a bit for cases where the same keys are being modified a lot.